qiling/core_hooks.py

```diff
@@ -312,6 +312,13 @@ def set_api(self, api_name, my_func):
         else:
             self.os.add_function_hook(api_name, my_func)
 
+    def set_partial_api(self, api_name, params: dict, my_func):
```

Comment:
the name is kind of bad. anyway we can merge into set api

btw, if we can do it for windows. we should be able to do it with UEFI at least ?

Reply:
We can't merge them together because in windows,  the function hook_winapi() that is the center of set_api(),  does not have the knowledge of the parameters

Reply:
I can give a look about uefi, but first I have to understand how things work there

Reply:
trow in a example and i will see what can i name

Reply:
my 2 cents
retain set_api, but expand a 3rd param and check if its a dict
```
def set_api(self, api_name, my_func, params=None):
    if isinstance(params, dict):
        if self.ostype == QL_OS.WINDOWS:
            params["func"] = my_func
            self.os.user_partial_defined_api[api_name] = params
        else:
            raise QlErrorCoreHook("Only windows is supported for now :(")
    else:
        if self.ostype == QL_OS.WINDOWS:
            self.os.user_defined_api[api_name] = my_func
        else:
            self.os.add_function_hook(api_name, my_func)
```